ocaml/xapi-aux/networking_info.mli

```diff
@@ -0,0 +1,33 @@
+(* Copyright (C) Citrix Systems Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation; version 2.1 only. with the special
+   exception on linking described in file LICENSE.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU Lesser General Public License for more details. *)
+
+val get_hostname : unit -> string
+(** [get_hostname ()] returns the hostname as returned by the command
+    [hostname]. The result may be cached. *)
+
+val reget_hostname : unit -> string
+(** [reget_hostname ()] returns the hostname as returned by the command
+    [hostname]. The result is never cached and the command is called every
+    time. *)
+
+exception Unexpected_address_type of string
+
+val hostnames : unit -> string list
```

Comment:
I think `dns_names` would be a better name for this function (my understanding is that we can have multiple FQDNs, but only one hostname)

Reply:
sure thing.
For reference, this is how the names look like in an 8.2 certificate:
`DNS:lcy2-dt119, DNS:lcy2-dt119.xenrtcloud`